src/NodeManager.php

```diff
@@ -86,7 +86,8 @@ private function generateBuffer(NodeModel $nodeModel): string
 
         if (null !== ($elements = $nodeModel->getContentElements())) {
             /** @var ContentModel $element */
-            foreach ($elements as $element) {
+            foreach ($elements as $key => $element) {
+                $element->index = $key;
```

Comment:
While we have magic properties in the Models, `index` is not represented in the data structure. This is modifying the model data and `$model->save()` would lead to an error.

Reply:
But how would it be possible to pass the sequence number to the individual element?

Reply:
Passing arrays to the template would also solve this problem 🙄